File: Grind169/Week1/232-Implement-Queue-using-Stacks/solution.py
class MyQueue:
    def __init__(self):
        # input and output are two stacks
        self.input = []
        self.output = []

    def push(self, x: int) -> None:
        self.input.append(x)

    def pop(self) -> int:
        if len(self.output) > 0:
            return self.output.pop()
        else:
            while len(self.input) > 0:
                self.output.append(self.input.pop())
            return self.output.pop()

    def peek(self) -> int:
        if len(self.output) > 0:
            return self.output[-1]
        else:
            while len(self.input) > 0:
                self.output.append(self.input.pop())
            return self.output[-1]

    def empty(self) -> bool:
        if len(self.input) == 0 and len(self.output) == 0:
            return True
        return False

File: Grind169/Week1/232-Implement-Queue-using-Stacks/solution.cpp
#include <stack>

using namespace std;

class MyQueue {
private:
    stack<int>input,output;

public:
    
    MyQueue() {
        
    }
    
    void push(int x) {
        input.push(x);
    }
    
    int pop() {
        int x;
        if (output.empty()==false){
            x= output.top();
            output.pop();
            return x;
        }
        else {
            while (input.empty()==false) {
                output.push(input.top());
                input.pop();
            }

            x=output.top();
            output.pop();
            return x;
        }
    }
    
    int peek() {
        if (output.empty()==false){
            return output.top();
        }
        else {
            while (input.empty()==false){
                output.push(input.top());
                input.pop();
            }

            return output.top();
        }
        
    }
    
    bool empty() {
        if (input.empty()&&output.empty()){
            return true;
        }
        return false;
    }
};